packages/payment-detection/src/erc777/superfluid-retriever.ts

```diff
@@ -0,0 +1,90 @@
+import { PaymentTypes } from '@requestnetwork/types';
+import { utils } from 'ethers';
+import { FlowUpdatedEvent } from '../thegraph/generated/graphql-superfluid';
+import { getTheGraphClient, TheGraphClient, TheGraphClientOptions } from '../thegraph/superfluid';
+
+/** Parameters for getting payment events from theGraph */
+type GraphPaymentQueryParams = {
+  reference: string;
+  to: string;
+  tokenAddress: string | null;
+};
+
+export class SuperFluidInfoRetriever {
+  private client: TheGraphClient;
+
+  /**
+   * @param paymentReference The reference to identify the payment
+   * @param tokenContractAddress The address of the ERC777 contract
+   * @param toAddress Address of the balance we want to check
+   * @param eventName Indicate if it is an address for payment or refund
+   * @param network The Ethereum network to use
+   * @param options Extra options to GraphQL client
+   */
+  constructor(
+    private paymentReference: string,
+    private tokenContractAddress: string | null,
+    private toAddress: string,
+    private eventName: PaymentTypes.EVENTS_NAMES,
+    private network: string,
+    private options?: TheGraphClientOptions,
+  ) {
+    this.client = getTheGraphClient(this.network, this.options);
+  }
+
+  private getGraphVariables(): GraphPaymentQueryParams {
+    return {
+      reference: utils.keccak256(`0x${this.paymentReference}`),
+      to: this.toAddress,
+      tokenAddress: this.tokenContractAddress,
+    };
+  }
+
+  public async getTransferEvents(): Promise<PaymentTypes.ERC20PaymentNetworkEvent[]> {
+    const variables = this.getGraphVariables();
+    const streamEvents = (await this.client.GetSuperFluidFlowEvents(variables)).flowUpdatedEvents;
+    // console.log('streamEvents:', streamEvents.length);
+
+    const untaggedEvents = (await this.client.GetSuperFluidUntaggedEvents(variables))
+      .flowUpdatedEvents;
+    // console.log('streamEvents:', streamEvents.length);
+
+    streamEvents.push(...untaggedEvents);
+    streamEvents.sort((a, b) => a.timestamp - b.timestamp);
```

Comment:
you should use the return of this method; it does not sort `streamEvents`, it creates a new sorted array. Besides, are you sure of the sort direction?

Reply:
I'm sure of the sort direction because when I've debugged step by step it has well sorted streamEvents in place without creating a new array variable.